### ISSUE TITLE
Error while spinning up

### ISSUE THREAD

**Issue by arimus** [Azure/vagrant-azure#108]
Having issues with the azure provider creating a resource group and the cloud service, but then bailing before creating the actual VMs or anything else.  The error I'm getting is below. This is output from a subsequent run, where it had already created the cloudservice, but the error is the same.

It seems to die on getting the image (5112500ae3b842c8b9c604889f8753c3__OpenLogic-CentOS-71-20150731), which it has to be finding in some capacity, because I get a different error if I tweak the name.
1. A more useful error message would be great
2. Any ideas what the problem might be?

Thanks in advance.

```
image = get_image(params[:image])
```

Full trace:

```
MyMac:test user$ vagrant up --provider=azure
Using provisioner ansible
==> default: Attempting to read state for testvagrant in testvagrant
Bringing machine 'default' up with 'azure' provider...
==> default: Attempting to read state for testvagrant in testvagrant
==> default: Determining OS Type By Image
==> default: OS Type is Linux
==> default: Attempting to read state for testvagrant in testvagrant
==> default: {:vm_name=>"testvagrant", :vm_user=>"vagrant", :image=>"5112500ae3b842c8b9c604889f8753c3__OpenLogic-CentOS-71-20150731", :location=>"West US"}
==> default: {:cloud_service_name=>"testvagrant", :storage_account_name=>"testvagrantstorag", :private_key_file=>"test.key", :vm_size=>"Small"}
==> default: Add Role? - false
Creating deployment...
Cloud service testvagrant already exists. Skipped...
Uploading certificate to cloud service testvagrant...
#
:
Failed to create virtual machine : :
/Users/user/.vagrant.d/gems/gems/azure-0.7.0/lib/azure/core/utility.rb:122:in `error_with_exit': Failed to create virtual machine : :  (RuntimeError)
    from /Users/user/.vagrant.d/gems/gems/azure-0.7.0/lib/azure/virtual_machine_management/virtual_machine_management_service.rb:137:in `rescue in create_virtual_machine'
    from /Users/user/.vagrant.d/gems/gems/azure-0.7.0/lib/azure/virtual_machine_management/virtual_machine_management_service.rb:96:in `create_virtual_machine'
    from /Users/user/.vagrant.d/gems/gems/vagrant-azure-1.3.0/lib/vagrant-azure/action/vagrant_azure_service.rb:38:in `method_missing'
    from /Users/user/.vagrant.d/gems/gems/vagrant-azure-1.3.0/lib/vagrant-azure/action/run_instance.rb:98:in `block in call'
    from /Users/user/.vagrant.d/gems/gems/vagrant-azure-1.3.0/lib/vagrant-azure/action/run_instance.rb:71:in `synchronize'
    from /Users/user/.vagrant.d/gems/gems/vagrant-azure-1.3.0/lib/vagrant-azure/action/run_instance.rb:71:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:95:in `block in finalize_action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builder.rb:116:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/runner.rb:66:in `block in run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/util/busy.rb:19:in `busy'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/runner.rb:66:in `run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builtin/call.rb:53:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
    from /Users/user/.vagrant.d/gems/gems/vagrant-azure-1.3.0/lib/vagrant-azure/action/os_type.rb:29:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
    from /Users/user/.vagrant.d/gems/gems/vagrant-azure-1.3.0/lib/vagrant-azure/action/connect_azure.rb:44:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builtin/config_validate.rb:25:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builtin/handle_box.rb:56:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
    from /Users/user/.vagrant.d/gems/gems/vagrant-triggers-0.5.2/lib/vagrant-triggers/action/trigger.rb:17:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
    from /Users/user/.vagrant.d/gems/gems/vagrant-triggers-0.5.2/lib/vagrant-triggers/action/trigger.rb:17:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
    from /Users/user/.vagrant.d/gems/gems/vagrant-triggers-0.5.2/lib/vagrant-triggers/action/trigger.rb:17:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/warden.rb:34:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/builder.rb:116:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/runner.rb:66:in `block in run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/util/busy.rb:19:in `busy'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/action/runner.rb:66:in `run'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/machine.rb:214:in `action_raw'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/machine.rb:191:in `block in action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/environment.rb:516:in `lock'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/machine.rb:178:in `call'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/machine.rb:178:in `action'
    from /opt/vagrant/embedded/gems/gems/vagrant-1.7.4/lib/vagrant/batch_action.rb:82:in `block (2 levels) in run'
    from /opt/vagrant/embedded/gems/gems/logging-1.8.2/lib/logging/diagnostic_context.rb:323:in `call'
    from /opt/vagrant/embedded/gems/gems/logging-1.8.2/lib/logging/diagnostic_context.rb:323:in `block in create_with_logging_context'
```

**Comment by arimus** [comment]
Forgot to add:

Vagrant 1.7.4
Vagrant Azure Plugin 1.3.0

**Comment by hongymagic** [comment]
I've also run into this issue today, with `--debug` argument:

```
Uploading certificate to cloud service xxx...
#
:
Failed to create virtual machine : :
ERROR warden: Error occurred: Failed to create virtual machine : :
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
ERROR warden: Error occurred: Failed to create virtual machine : :
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
ERROR warden: Error occurred: Failed to create virtual machine : :
 INFO warden: Beginning recovery process...
 INFO warden: Calling recover: #<Vagrant::Action::Builtin::Call:0x00000102611ab8>
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO warden: Beginning recovery process...
 INFO warden: Recovery complete.
 INFO environment: Released process lock: machine-action-b3fc63ff97a8ce82eef54647681936d9
 INFO environment: Running hook: environment_unload
 INFO runner: Preparing hooks for middleware sequence...
 INFO runner: 1 hooks defined.
…
```

**Comment by devigned** [comment]
For Linux boxes, I think you will find the 2.0 preview release is much easier to use.
